lib/streamlit/UploadedFileManager.py

```diff
@@ -53,39 +44,45 @@ class UploadedFileManager(object):
     """
 
     def __init__(self):
-        self._files = {}
+        self._files_by_id = {}  # type: Dict[Tuple[str, str], UploadedFileList]
         self._files_lock = threading.Lock()
```

Comment:
Can we document what this lock does?

Reply:
You know - I think it's actually not necessary, since we're just using a primitive dict here. All dict operations are themselves atomic, and we don't have an instance where we need to do multiple dict operations atomically. So I'm going to remove it!

Reply:
Ah wait - no, `dict.keys()` is not thread-safe (an exception is thrown if you mutate while iterating over the keys). So I'll keep the lock and provide a comment!

Reply:
Thanks!